.github/scripts/check_task_and_pipeline_yamls.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+shopt -s nullglob
+set -euo pipefail
+
+echo ">>> Apply tasks"
+for task_folder in task/*/; do
+  if [ -d "$task_folder" ]; then
+    task="$(basename "$task_folder")"
+    echo ">>> Task: $task"
+    (
+      cd "$task_folder"
+      for version in */; do
+        if [ -d "$version" ]; then
+          kubectl apply -f "$version/$task.yaml" --dry-run=server
+        fi
+      done
+    )
+  fi
+done
+
+echo ">>> Apply pipelines"
+
+cd pipelines
+
+ignored_pipelines=(
+  "template-build"
+)
+
+for pipeline in */; do
+  if [ -d "$pipeline" ]; then
+    pipeline="$(basename "$pipeline")"
+    for ignored in "${ignored_pipelines[@]}"; do
+      if [ "$ignored" == "$pipeline" ]; then
+        echo ">>> Ignoring pipeline: $pipeline"
+        continue 2
+      fi
+    done
+    echo ">>> Pipeline: $pipeline"
+    to_apply=()
+    if [ -e "$pipeline/kustomization.yaml" ]; then
+      kubectl kustomize "$pipeline" > "$pipeline/generated_pipeline.yaml"
+      to_apply+=("$pipeline/generated_pipeline.yaml")
+    else
+      for yaml in "$pipeline"/*.yaml; do
+        if [ -f "$yaml" ]; then
+          to_apply+=("$yaml")
+        fi
+      done
```

Comment:
You're right that not all pipelines follow the naming convention :disappointed: 

But we do have the same kustomization checks as we do for tasks - the generated file should always be committed to the repo and up to date, so it shouldn't be necessary to re-run kustomize. What do you think about doing it this way:

for each YAML file in directory, if `.kind == "Pipeline"`, validate the file

Reply:
Done